src/Swarm/Game/Step.hs

```diff
@@ -1627,13 +1641,13 @@ execConst c vs s k = do
             missingParentInv = neededParentInv `E.difference` parentInventory
             missingMap =
               M.fromList
-                . filter ((> 0) . snd)
+                . filter (\(n, x) -> x > 0 && (x /= PosInfinity || countByName n parentInventory /= PosInfinity))
```

Comment:
Can you walk me through the logic here?  I'm not sure I follow.

Reply:
To be safe removing an infinite count of items from an infinite count of items is still an infinite count of items.

But here we were being smart by removing the _robot inventory_ from _required items_.

So the infinity stays and is not 0.

It is a hack to keep the changes minimal, but we should eventually clean up the code by changing the data structures.
I will write my thoughts on that in a separate comment.